File: OOYA_SP/Java/src/main/java/com/ran/splabjava/Programmer.java
package com.ran.splabjava;

import java.util.List;

abstract public class Programmer implements Comparable<Programmer> {

    private String name;
    private String surname;
    private Position position;
    private List<String> skills;

    public Programmer(String name, String surname, Position position, List<String> skills) {
        this.name = name;
        this.surname = surname;
        this.position = position;
        this.skills = skills;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public Position getPosition() {
        return position;
    }

    public void setPosition(Position position) {
        this.position = position;
    }

    public List<String> getSkills() {
        return skills;
    }

    public void setSkills(List<String> skills) {
        this.skills = skills;
    }

    abstract void print();

    public String getSkillsAsString() {
        return skills.stream()
                .reduce((first, second) -> first + ", " + second)
                .get();
    }

    public int compareTo(Programmer other) {
        if (this.position != other.position) {
            return Integer.compare(this.position.ordinal(), other.position.ordinal());
        } else if (!this.surname.equals(other.surname)) {
            return this.surname.compareTo(other.surname);
        } else {
            return this.name.compareTo(other.name);
        }
    }

}

File: OOYA_SP/Python/Programmer.py
from Position import Position
class Programmer: #Базовый класс
    def __init__(self, nm, srnm, pos, skls):
        self._name = nm
        self._surname = srnm
        self._position = pos
        self._skills = skls

    def getName(self):  #иммитация публичного свойства, на самом деле все свойства в питоне паблик
        return self._name

    def setName(self, val):# set метод может быть вызван классом наследником
         self._name = val

    def __print__(self):
        print("Programmer", self._name + ' has skills:' + self._skills)




